main.go

```diff
@@ -268,37 +269,46 @@ func main() {
 			return perrors.Wrapf(err, "unable to create remote cluster cache tracker")
 		}
 
-		// Check for non-supervisor VSphereCluster and start controller if found
-		gvr := infrav1.GroupVersion.WithResource(reflect.TypeOf(&infrav1.VSphereCluster{}).Elem().Name())
-		isNonSupervisorCRDLoaded, err := isCRDDeployed(mgr, gvr)
-		if err != nil {
-			return err
+		govimomiGVR := infrav1.GroupVersion.WithResource(reflect.TypeOf(&infrav1.VSphereCluster{}).Elem().Name())
+		supervisorGVR := vmwarev1.GroupVersion.WithResource(reflect.TypeOf(&vmwarev1.VSphereCluster{}).Elem().Name())
+
+		var isSupervisorCRDLoaded, isGovimoniCRDLoaded bool
+		if err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 30*time.Second, true, func(ctx context.Context) (bool, error) {
+			var errGovimomi, errSupervisor error
+
+			// Check for non-supervisor VSphereCluster and start controller if found
+			isGovimoniCRDLoaded, errGovimomi = isCRDDeployed(mgr, govimomiGVR)
+
+			// Check for supervisor VSphereCluster and start controller if found
+			isSupervisorCRDLoaded, errSupervisor = isCRDDeployed(mgr, supervisorGVR)
+
+			if errGovimomi != nil || errSupervisor != nil {
```

Comment:
Not sure if we should change that logic. Was talking about this with Christian this morning and depending on the exact state of the cluster we can hit issues that we don't cover (e.g. if the apiGroup exists but not the VSphereCluster CRD).

What about changing this to return if one of the (loaded, error) pairs is (true, nil)

Reply:
> Not sure if we should change that logic

That might have been misleading. I didn't mean that we made a change compared to the previous version. More that we should consider changing the current version on the PR :)